src/components/button/button.ts

```diff
@@ -48,6 +57,18 @@ export class MdButton {
     TimerWrapper.setTimeout(() => { this.isMouseDown = false; }, 100);
   }
 
+  _updateColor(newColor: string) {
+    if (this._color != null && this._color != '') {
+      this.renderer.setElementClass(this.elementRef.nativeElement, `md-${this._color}`, false);
+    }
+
+    if (newColor != null && newColor != '') {
+      this.renderer.setElementClass(this.elementRef.nativeElement, `md-${newColor}`, true);
+    }
```

Comment:
There's some repetition between lines 61-63 and lines 65-67. Can you refactor to share more code, like:

```javascript
set color(value: string) {
   this._updateColor(this._color, false);
   this._updateColor(value, true);
}

_updateColor(color: string, bool: boolean): void {
   if (color != null && color != '') {
      this.renderer.setElementClass(this.elementRef.nativeElement, `md-${color}`, bool);
   }
}
```